MapboxCoreNavigation/SimulatedLocationManager.swift

```diff
@@ -88,14 +91,18 @@ open class SimulatedLocationManager: NavigationLocationManager {
     @objc public init(routeProgress: RouteProgress) {
         super.init()
         let currentDistance = calculateCurrentDistance(routeProgress.distanceTraveled)
-        initializeSimulatedLocationManager(for: routeProgress.route, currentDistance: currentDistance, currentSpeed: 0)
+        commonInit(for: routeProgress.route, currentDistance: currentDistance, currentSpeed: 0)
     }
 
-    private func initializeSimulatedLocationManager(for route: Route, currentDistance: CLLocationDistance, currentSpeed: CLLocationSpeed) {
+    private func commonInit(for route: Route, currentDistance: CLLocationDistance, currentSpeed: CLLocationSpeed) {
         
         self.currentSpeed = currentSpeed
         self.currentDistance = currentDistance
         self.route = route
+
+        self.timer = DispatchTimer(countdown: .milliseconds(0), repeating: updateInterval, accuracy: accuracy, executingOn: .main) { [weak self] in
+            DispatchQueue.main.async { self?.tick() }
```

Comment:
Is this async call necessary if we're `executingOn: .main`?

Reply:
if you want to defer to a later run loop iteration ...